src/main/java/whobot/utils/TaskList.java

```diff
@@ -0,0 +1,293 @@
+                    + " The description can not be empty.");
+        }
+    }
+
+    /***
+     * Adds a Event Task to the Task List
+     *
+     * @param command the command with the details for the task
+     * @param ui UI to show output to
+     * @throws WhoBotException If there is any error on addition
+     */
+    public void addEvent(String command, UI ui) throws WhoBotException {
+        try {
+            if (!command.contains("/at ")) {
+                throw new WhoBotException("Ensure that the command is of the form \"event #description /at #timing\"."
+                        + " The timing must be given.");
+            }
+            Event task = new Event(command.substring(6));
+            if (list.add(task)) {
+                ui.echo("I have added this Event Task to the list: \"" + task.getDescription() + "\"", UI.Type.START);
+                ui.echo("You now have " + list.size() + " task(s) in the list.", UI.Type.END);
+            } else {
+                throw new WhoBotException("I am sorry. The task couldn't be added, please try again.");
+            }
+        } catch (IndexOutOfBoundsException e) {
+            throw new WhoBotException("Ensure that the command is of the form \"event #description /at #timing\"."
+                    + " The description can not be empty.");
+        }
+    }
+
+    /***
+     * Adds a Deadline Task to the Task List
+     *
+     * @param command the command with the details for the task
+     * @param ui UI to show output to
+     * @throws WhoBotException If there is any error on addition
+     */
+    public void addDeadline(String command, UI ui) throws WhoBotException {
+        try {
+            if (!command.contains("/by ")) {
+                throw new WhoBotException("Ensure that the command is of the form "
+                        + "\"deadline #description /by #deadline\"."
+                        + " The deadline must be given.");
+            }
+            Deadline task = new Deadline(command.substring(9));
+            if (list.add(task)) {
+                ui.echo("I have added this Deadline Task to the list: \"" + task.getDescription() + "\"",
+                        UI.Type.START);
+                ui.echo("You now have " + list.size() + " task(s) in the list.", UI.Type.END);
+            } else {
+                throw new WhoBotException("I am sorry. The task couldn't be added, please try again.");
+            }
+        } catch (IndexOutOfBoundsException e) {
+            throw new WhoBotException("Ensure that the command is of the form \"deadline #description /by #deadline\"."
+                    + " The description can not be empty.");
+        }
+    }
+
+    /***
+     * Shows Tasks on a specific date
+     * @param command the command with the date specified
+     * @param ui UI to show output to
+     * @throws WhoBotException If there is any error in the command
+     */
```

Comment:
Can there be an empty line between description and parameter section?
```suggestion
    /***
     * Shows Tasks on a specific date
     *
     * @param command the command with the date specified
     * @param ui UI to show output to
     * @throws WhoBotException If there is any error in the command
     */
```